Assets/MRTK/SDK/Features/UX/Scripts/Utilities/PrefabSpawner.cs

```diff
@@ -0,0 +1,249 @@
+        [SerializeField]
+        private VanishType vanishType = VanishType.VanishOnFocusExit;
+        [SerializeField]
+        private RemainType remainType = RemainType.Timeout;
+        [Header("Timing")]
+        [SerializeField]
+        [Range(0f, 5f)]
+        private float appearDelay = 0.0f;
+        [SerializeField]
+        [Range(0f, 5f)]
+        private float vanishDelay = 2.0f;
+        [SerializeField]
+        [Range(0.5f, 10.0f)]
+        private float lifetime = 1.0f;
+        [Header("Orientation")]
+        [SerializeField]
+        private bool keepWorldRotation = true;
+
+        private float focusEnterTime = 0f;
+        private float focusExitTime = 0f;
+        private float tappedTime = 0f;
+
+        private GameObject spawnable;
+
+        private async void ShowSpawnable()
+        {
+            await UpdateSpawnable(focusEnterTime, tappedTime);
+        }
+
+        private async Task UpdateSpawnable(float focusEnterTimeOnStart, float tappedTimeOnStart)
+        {
+            if (appearType == AppearType.AppearOnFocusEnter)
+            {
+                if (spawnable == null)
+                {
+                    spawnable = Instantiate(prefab, transform, false);
+                    spawnable.transform.localPosition = Vector3.zero;
+                    if (!keepWorldRotation)
+                    {
+                        spawnable.transform.localRotation = Quaternion.identity;
+                    }
+                    spawnable.gameObject.SetActive(false);
+                }
+                // Wait for the appear delay
+                await new WaitForSeconds(appearDelay);
+                // If we don't have focus any more, get out of here
+
+                if (!HasFocus)
+                {
+                    return;
+                }
+            }
+
+            spawnable.gameObject.SetActive(true);
+
+            SpawnableActivated(spawnable);
+
+            while (spawnable.gameObject.activeSelf)
+            {
+                if (remainType == RemainType.Timeout)
+                {
+                    switch (appearType)
+                    {
+                        case AppearType.AppearOnTap:
+                            if (Time.unscaledTime - tappedTime >= lifetime)
+                            {
+                                spawnable.gameObject.SetActive(false);
+                                return;
+                            }
+
+                            break;
+                        case AppearType.AppearOnFocusEnter:
+                            if (Time.unscaledTime - focusEnterTime >= lifetime)
+                            {
+                                spawnable.gameObject.SetActive(false);
+                                return;
+                            }
+
+                            break;
+                    }
+                }
+
+                //check whether we're suppose to disappear
```

Comment:
```suggestion
                // Check whether we're supposed to disappear
```

Reply:
Fixes the PR validation checker blocking on a comment without a space at the beginning